### PR TITLE
Add support for Spot instances

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it

Towards https://github.com/giantswarm/giantswarm/issues/27925

### Checklist

- [X] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-aws/values.schema.json [hunk 1]
```diff
@@ -76,6 +76,23 @@
                     "type": "integer",
                     "title": "Root volume size (GB)"
                 },
+                "spotInstances": {
+                    "type": "object",
+                    "title": "Spot VM instances",
+                    "description": "Spot VM instances configuration",
```
Comment (marians) [inline helm/cluster-aws/values.schema.json:82]: AWS calls them "spot instances", without the "VM".

Also trying to make the description more useful.

```suggestion
                    "title": "Spot instances",
                    "description": "Compared to on-demand instances, spot instances can help you save cost.",
```

File: helm/cluster-aws/values.schema.json [hunk 2]
```diff
@@ -76,6 +76,23 @@
...
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "title": "Enable",
+                            "default": false
+                        },
+                        "maxPrice": {
+                            "type": "string",
+                            "title": "Maximum price to pay for Spot VM instances",
```
Comment (marians) [inline helm/cluster-aws/values.schema.json:91]:
```suggestion
                            "title": "Maximum price to pay per instance per hour, in USD.",
```

File: helm/cluster-aws/values.schema.json [hunk 3]
```diff
@@ -76,6 +76,23 @@
...
+                    "title": "Spot VM instances",
+                    "description": "Spot VM instances configuration",
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "title": "Enable",
+                            "default": false
+                        },
+                        "maxPrice": {
+                            "type": "string",
```
Comment (marians) [inline helm/cluster-aws/values.schema.json:90]: I'm thinking this should be a `number` type field.

Comment (fiunchinho) [reply helm/cluster-aws/values.schema.json]: Done

File: helm/cluster-aws/values.schema.json (hunk 2 above)
Comment (fiunchinho) [reply helm/cluster-aws/values.schema.json]: Done

File: helm/cluster-aws/values.schema.json (hunk 1 above)
Comment (fiunchinho) [reply helm/cluster-aws/values.schema.json]: Done

**Review by calvix — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add support for Spot VM instances.
```
Comment (marians) [inline CHANGELOG.md:12]:
```suggestion
- Add support for Spot instances.
```

**Pull request merged**